File: 프로그래머스/1/250121. ［PCCE 기출문제］ 10번 ／ 데이터 분석/［PCCE 기출문제］ 10번 ／ 데이터 분석.java
import java.util.*;
class Solution {
    public int[][] solution(int[][] data, String ext, int val_ext, String sort_by) {
        String[] extArray = new String[]{"code", "date", "maximum", "remain"};
        int index = 0;
        for(int i=0; i<4; i++){
            if(ext.equals(extArray[i])){
                index = i;
            }
        }
        List<Integer> indexList = new ArrayList<>();
        for(int i=0; i<data.length; i++){
            if(data[i][index] < val_ext){
                indexList.add(i);
            }
        }
        for(int i=0; i<4; i++){
            if(sort_by.equals(extArray[i])){
                index = i;
            }
        }
        List<Integer> sortList = new ArrayList<>();
        for(int i=0; i<indexList.size(); i++){
            sortList.add(data[indexList.get(i)][index]);
        }
        Collections.sort(sortList);
        List<Integer> list = new ArrayList<>();
        for(int i=0; i<sortList.size(); i++){
            for(int j=0; j<indexList.size(); j++){
                if(data[indexList.get(j)][index] == sortList.get(i)){
                    list.add(indexList.get(j));
                }
            }
        }
        int[][] answer = new int[list.size()][data[0].length];
        for(int i=0; i<list.size(); i++){
            for(int j=0; j<data[0].length; j++){
                answer[i][j] = data[list.get(i)][j];
            }
        }
        return answer;
    }
}

File: 프로그래머스/1/250121. ［PCCE 기출문제］ 10번 ／ 데이터 분석/［PCCE 기출문제］ 10번 ／ 데이터 분석.py
def solution(data, ext, val_ext, sort_by):
    extArray = ["code", "date", "maximum", "remain"]
    index = 0
    for i in range(4) :
        if ext == extArray[i] :
            index = i
    
    indexList = []
    for i in range(len(data)) :
        if data[i][index] < val_ext :
            indexList.append(i)
    
    for i in range(4) :
        if sort_by == extArray[i] :
            index = i
            
    sortList = []
    for i in range(len(indexList)) :
        sortList.append(data[indexList[i]][index])
    
    sortList.sort()
    
    list = []
    for i in range(len(sortList)) :
        for j in range(len(indexList)) :
            if data[indexList[j]][index] == sortList[i] :
                list.append(indexList[j])
    
    answer = [[0] * len(data[0]) for _ in range(len(list))]
    for i in range(len(sortList)) :
        for j in range(len(indexList)) :
            if data[indexList[j]][index] == sortList[i] :
                for k in range(len(data[0])) :
                    answer[i][k] = data[indexList[j]][k]

    return answer
